### PR TITLE
Wrapped crypto-js with the CRT hash interface, added unit testing

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/browser/crypto.ts [hunk 1]
```diff
@@ -13,4 +13,48 @@
...
+        const encoder = new TextEncoder();
+        const bytes = encoder.encode(truncated);
+        return new DataView(bytes);
+    }
+}
+
+export function hash_md5(data: Hashable, truncate_to?: number): DataView {
+    const md5 = new Md5Hash();
+    md5.update(data);
+    return md5.digest();
```
Comment (bretambrose) [inline lib/browser/crypto.ts]: Does truncate_to get magically used here?

File: lib/browser/crypto.ts [hunk 2]
```diff
@@ -13,4 +13,48 @@
...
+import { Hashable } from "../common/crypto";
+
+export class Md5Hash {
+    private hash?: Crypto.WordArray;
+    
+    update(data: Hashable) {
+        this.hash = Crypto.MD5(data.toString(), this.hash ? this.hash.toString() : undefined);
+    }
+
+    digest(truncate_to?: number): DataView {
```
Comment (bretambrose) [inline lib/browser/crypto.ts]: I have to confess I like "finalize" better than "digest" both for consistency as well as action vs. noun.

File: lib/browser/crypto.ts [hunk 3]
```diff
@@ -13,4 +13,48 @@
...
+    }
+}
+
+export function hash_md5(data: Hashable, truncate_to?: number): DataView {
+    const md5 = new Md5Hash();
+    md5.update(data);
+    return md5.digest();
+}
+
+export class Sha256Hmac {
```
Comment (bretambrose) [inline lib/browser/crypto.ts]: Does this need a top-level wrapper function?

File: lib/browser/crypto.ts (hunk 1 above)
Comment (justinboswell) [reply lib/browser/crypto.ts]: No, it gets magically forgotten by the implementer, thanks!

File: lib/browser/crypto.ts (hunk 2 above)
Comment (justinboswell) [reply lib/browser/crypto.ts]: Now is the time, I can fix it!

File: lib/browser/crypto.ts (hunk 3 above)
Comment (justinboswell) [reply lib/browser/crypto.ts]: yes it does, I deleted it by accident, aaaahhhhh

File: tsconfig.browser.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
     },
     "include": [
         "lib/*.ts",
```
Comment (ColdenCullen) [inline tsconfig.browser.json]: This should be empty now, right?

File: tsconfig.json [hunk 5]
```diff
@@ -2,6 +2,7 @@
     "extends": "./tsconfig.base",
     "include": [
         "lib/*.ts",
```
Comment (ColdenCullen) [inline tsconfig.json]: Same

File: tsconfig.browser.json (hunk 4 above)
Comment (justinboswell) [reply tsconfig.browser.json]: Fixed.

File: tsconfig.json (hunk 5 above)
Comment (justinboswell) [reply tsconfig.json]: Fixed

**Pull request merged**